engine/common/miniz.h

```diff
@@ -198,15 +198,15 @@ extern "C" {
 typedef unsigned long mz_ulong;
 
 /* mz_free() internally uses the MZ_FREE() macro (which by default calls free() unless you've modified the MZ_MALLOC macro) to release a block allocated from the heap. */
-void mz_free(void *p);
+static void mz_free(void *p);
```

Comment:
Instead of re-defining them as static, you could define `MINIZ_HEADER_FILE_ONLY` before including miniz. 
miniz implementation already defined in filesystem.c, but if you want, you can move it to dummy file.

Reply:
I want to refactor this library anyway.

Reply:
I have generated it as a single-header in hope that it will not change, except obvious updates. :)